echo-webhooks/src/test/groovy/com/netflix/spinnaker/echo/controllers/WebhooksControllerSpec.groovy

```diff
@@ -153,129 +143,45 @@ class WebhooksControllerSpec extends Specification {
 
     response.eventProcessed == true
     response.eventId == event.eventId
+    event.content.branch == "refs/heads/master"
+    event.content.repoProject == "repository"
+    event.content.slug == "repository"
+    event.content.hash == "178864a7d521b6f5e720b386b2c2b0ef8563e0dc"
   }
 
   void 'handles Bitbucket Server Merge Webhook'() {
     def event
     given:
-    WebhooksController controller = new WebhooksController(mapper: new ObjectMapper())
+    WebhooksController controller = new WebhooksController(mapper: new ObjectMapper(), scmWebhookHandler: scmWebhookHandler)
     controller.propagator = Mock(EventPropagator)
     controller.artifactExtractor = Mock(ArtifactExtractor)
     controller.artifactExtractor.extractArtifacts(_, _, _) >> []
+    HttpHeaders headers = new HttpHeaders();
+    headers.add("X-Event-Key", "pr:merged")
+
 
     when:
     def response = controller.forwardEvent(
       'git',
       'bitbucket',
       '''{
   "eventKey":"pr:merged",
-  "date":"2017-09-19T10:39:36+1000",
```

Comment:
Why are these tests getting deleted?

Reply:
The reason I used the whole request was
1. paranoia - so more sideeffect are accounted for
2. reference - these are the examples given by Bitbucket themselves. So it should always work.

Reply:
Valid points, will add them back!